x/erc20/keeper/ibc_callbacks.go

```diff
@@ -125,7 +121,7 @@ func (k Keeper) OnRecvPacket(
 		return ack
 
 	// Case 2. native ERC20 token
-	case pair.IsNativeERC20():
+	case found && pair.IsNativeERC20():
```

Comment:
nit: isn't this `found` redundant given the first case?

Reply:
if its not found `pair` will have an unknownOwner and not be NativeErc20.
But I think this is a little bit more verbose